frontend/javascripts/oxalis/model/accessors/dataset_accessor.js

```diff
@@ -538,31 +562,111 @@ export function isColorLayer(dataset: APIDataset, layerName: string): boolean {
   return getLayerByName(dataset, layerName).category === "color";
 }
 
-export function getSegmentationLayer(dataset: APIMaybeUnimportedDataset): ?APISegmentationLayer {
+export function getVisibleSegmentationLayer(state: OxalisState): ?APISegmentationLayer {
+  const { datasetConfiguration } = state;
+  const { viewMode } = state.temporaryConfiguration;
+
+  const visibleSegmentationLayers = getSegmentationLayers(state.dataset).filter(layer =>
+    isLayerVisible(state.dataset, layer.name, datasetConfiguration, viewMode),
+  );
+  if (visibleSegmentationLayers.length > 0) {
+    return visibleSegmentationLayers[0];
+  }
+
+  return null;
+}
+
+export function getVisibleSegmentationLayers(state: OxalisState): Array<APISegmentationLayer> {
+  const { datasetConfiguration } = state;
+  const { viewMode } = state.temporaryConfiguration;
+  const segmentationLayers = getSegmentationLayers(state.dataset);
+  const visibleSegmentationLayers = segmentationLayers.filter(layer =>
+    isLayerVisible(state.dataset, layer.name, datasetConfiguration, viewMode),
+  );
+  return visibleSegmentationLayers;
+}
+
+export function getSegmentationLayerWithMappingSupport(state: OxalisState): ?APISegmentationLayer {
+  // From the visible segmentation layers, return the first layer which has enabled mappings.
+  // If no layer has enabled mappings, pick a layer which has some mappings (this is important
+  // for the initialization of the mapping textures, since isMappingEnabled will be set to true
+  // after all mapping data was copied to the GPU, but getSegmentationLayerWithMappingSupport is
+  // already used before that to prepare the mapping for the correct layer).
+  // Currently, webKnossos only supports one active mapping at a given time. The UI should ensure
+  // that not more than one mapping is enabled.
+
+  const visibleSegmentationLayers = getVisibleSegmentationLayers(state);
+
+  const layersWithEnabledMapping = visibleSegmentationLayers.filter(layer => {
+    const mappingInfo = state.temporaryConfiguration.activeMappingByLayer[layer.name];
+    return mappingInfo && mappingInfo.isMappingEnabled;
+  });
+
+  if (layersWithEnabledMapping.length > 0) {
+    return layersWithEnabledMapping[0];
+  }
+  const layersWithMappings = visibleSegmentationLayers.filter(
+    layer => layer.mappings && layer.mappings.length > 0,
```

Comment:
Just to make sure, does this include agglomerate file mappings?

Reply:
Good point! I just checked it and agglomerates are stored separately in `layer.agglomerates` (which fits the implementation of this function well, since agglomerates aren't handled by the GPU).